target/linux/ath79/image/generic.mk

```diff
@@ -999,6 +999,21 @@ define Device/librerouter_librerouter-v1
 endef
 TARGET_DEVICES += librerouter_librerouter-v1
 
+define Device/meraki_mr16
+  SOC := ar7161
+  DEVICE_VENDOR := Meraki
+  DEVICE_MODEL := MR16
+  ROOTFS_SIZE := 13440k
+  IMAGE_SIZE := 15680k
+  DEVICE_PACKAGES := kmod-spi-gpio kmod-owl-loader
+  IMAGE/kernel.bin := append-kernel
+  IMAGE/rootfs.bin := append-rootfs | pad-rootfs
```

Comment:
What are these two images good for? They are not mentioned in the flashing instructions.

Reply:
I need to rewrite the commit; sadly the flashing process listed there is inaccurate for an OEM MR16. The user must also set u-boot variables for where to boot from by default; in the past these images were used for initial installation, but that may no longer be necessary with a dts rather than a mach. I'm not sure. I'll test again from an OEM and rewrite the instructions.

https://github.com/btvmeshnet/openwrt/pull/1#discussion_r456921586 for reference.

Reply:
I was additionally startled by the fact that you just define a single firmware partition now, but still do have a fixed offset for append-kernel in sysupgrade recipe. I wonder whether that breaks at some point, or will be irrelevant.

Reply:
I see this. I've unraveled the firmware partition in the .dts into a rootfs and kernel partition, and restructured the image builder to provide for a fixed offset for the kernel so that the instructions are applicable without conflicting with the dts.

Reply:
As stated elsewhere already, I wonder whether this "rootfs first" approach is actually, or whether we could just use a single firmware partition with one of the splitters instead.

Reply:
I think the reason for the "rootfs first" approach is because the kernel offset has to be fixed for u-boot, but I'm not sure.

Reply:
And that's why I was startled by you just moving that +64k for adding the mac partition at 0x80000.

Reply:
I've followed your lead here -- the kernel is first as it should be, and we preserve the Meraki config partition, and now have a tested (but not officially supported) solution for rearranging things as needed to migrate ar71xx to ath79 without opening the case, so I'd like to call this one done.